runtime/memory/64bit_nix.c

```diff
@@ -37,6 +37,19 @@ void expand_memory() {
     memory_size += WASM_PAGE_SIZE;
 }
 
+i32 instruction_memory_size() {
+    return memory_size / WASM_PAGE_SIZE;
+}
+
+i32 instruction_memory_grow(i32 count) {
+    i32 prev_size = instruction_memory_size();
+    for (int i = 0; i < count; i++) {
+        expand_memory();
+    }
+
+    return prev_size;
+}
+
 INLINE char* get_memory_ptr_for_runtime(u32 offset, u32 bounds_check) {
     // Due to how we setup memory for x86, the virtual memory mechanism will catch the error, if bounds < WASM_PAGE_SIZE
     assert(bounds_check < WASM_PAGE_SIZE || (memory_size > bounds_check && offset <= memory_size - bounds_check));
```

Comment:
Put another way: we might want to (in the future) create an api for sandbox termination. This would enable environments with more than one sandbox in the process (e.g. sledge)

Reply:
@gparmer I think we'd need to add special handling in `aWsm` for such a use case, so maybe we could do a different API? Creating branches manually in `aWsm` is tricky, so it might be better to redo a different memory API for that use case